src/hooks/useApproveAndDeposit.ts

```diff
@@ -107,10 +109,18 @@ export function useApproveAndDeposit(
         gasPriceUnsafe ? String(gasPriceUnsafe) : "45",
         9,
       )
-      const approveSingleToken = async (token: Token): Promise<void> => {
+      const approveSingleToken = async (
+        token: BasicToken | undefined,
+      ): Promise<void> => {
+        if (!token || !library) return
```

Comment:
Would prefer to give the user a visible error here since this fn is called when the user clicks a button.